admin/data/sql/migrations/148.do.create-stored-procedure-to-create-checks.sql

```diff
@@ -0,0 +1,89 @@
+-- create the create checks stored procedure
+DROP PROCEDURE  IF EXISTS [mtc_admin].[spCreateChecks];
+GO
+CREATE PROCEDURE [mtc_admin].[spCreateChecks]
+    @TVP [mtc_admin].[CheckTableType] READONLY
+AS
+
+  -- Connection pooling is enabled - make sure we are not in an existing transaction
+  IF @@TRANCOUNT <> 0  -- Rollback old transactions before starting another
+    ROLLBACK TRANSACTION
+
+  BEGIN TRY
+  BEGIN TRANSACTION
+
+  DECLARE checkArgsList CURSOR
+  FOR SELECT pupil_id, checkForm_id, checkWindow_id, isLiveCheck, pinExpiresAt, school_id
+      FROM @TVP
+    FOR READ ONLY
+
+  DECLARE @pupilId int
+  DECLARE @checkFormId int
+  DECLARE @checkWindowId int
+  DECLARE @isLiveCheck bit
+  DECLARE @pinExpiresAt datetimeoffset
+  DECLARE @schoolId int
+  DECLARE @pin int
+  DECLARE @err int
+  DEClARE @checkId int
+  DECLARE @output TABLE (id int);
+
+  OPEN checkArgsList
+  FETCH checkArgsList INTO @pupilId, @checkFormId, @checkWindowId, @isLiveCheck, @pinExpiresAt, @schoolId
+  WHILE (@@FETCH_STATUS = 0) BEGIN
+
+    -- Create the check
+    INSERT INTO [mtc_admin].[check]
+        (pupil_id, checkForm_id, checkWindow_id, isLiveCheck, pinExpiresAt, school_id)
+    VALUES (@pupilId, @checkFormId, @checkWindowId, @isLiveCheck, @pinExpiresAt, @schoolId);
+
+    -- Get the check.id we just inserted
+    SET @checkId = SCOPE_IDENTITY();
+
+    -- Assign a pin to the check
+    INSERT INTO [mtc_admin].[checkPin] (school_id, check_id, pin_id)
+    VALUES (@schoolId, @checkId, (SELECT TOP 1 p.id
+                                  FROM [mtc_admin].[pin] p
+                                         LEFT JOIN [mtc_admin].[checkPin] cp ON (p.id = cp.pin_id)
+                                  WHERE cp.pin_id IS NULL
+                                    AND (cp.school_id IS NULL OR cp.school_id <> 1)));
```

Comment:
i don't believe we should hardcode this school_id of `1`, as discussed

Reply:
Fixed